src/applications/disability-benefits/all-claims/tests/components/ComboBox.unit.spec.jsx

```diff
@@ -36,111 +40,375 @@ describe('ComboBox', () => {
+      expect(listbox).to.have.length(0);
+    });
 
-      expect(input.value).to.eq(filteredItems[0]);
+    it('should retain input value and make listbox empty when click outside', () => {
+      const searchTerm = 'f';
+      const { getByRole, getByTestId } = render(<ComboBox {...props} />);
 
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
       const listbox = getByRole('listbox');
+
+      expect(listbox).to.have.length(21);
+
+      fireEvent.click(document);
+
+      expect(input).to.have.value(searchTerm);
       expect(listbox).to.have.length(0);
     });
   });
+
+  describe('Keyboard Interactions', () => {
+    it('should highlight listbox items down the list on ArrowDown', () => {
+      const searchTerm = 'g';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listboxItems = getAllByRole('option');
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      expect(listboxItems[0]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[0]).to.have.attribute('aria-selected', 'true');
+
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      expect(listboxItems[1]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[1]).to.have.attribute('aria-selected', 'true');
+    });
+
+    it('should stop at the last item in the listbox on repeated ArrowDown', () => {
+      const searchTerm = 'h';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listboxItems = getAllByRole('option');
+
+      for (let i = 0; i < 24; i++) {
+        fireEvent.keyDown(input, { key: 'ArrowDown' });
+      }
+
+      expect(listboxItems[20]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[20]).to.have.attribute('aria-selected', 'true');
+    });
+
+    it('should highlight listbox items up the list on ArrowUp', () => {
+      const searchTerm = 'i';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listboxItems = getAllByRole('option');
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      expect(listboxItems[1]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[1]).to.have.attribute('aria-selected', 'true');
+
+      fireEvent.keyDown(input, { key: 'ArrowUp' });
+
+      expect(listboxItems[0]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[0]).to.have.attribute('aria-selected', 'true');
+    });
+
+    // TODO: This test is working in the browser but here it's getting stuck on the free-text item
```

Comment:
I tried rewriting this test with `userEvent` and `waitFor` and the focus didn't get stuck on the "Enter as condition" option, but it DID move focus to the body 

`    it('should move focus to input on ArrowUp when at the top of the list', async () => {
      const searchTerm = 'i';
      const { getByTestId } = render(<ComboBox {...props} />);
      
      const input = getByTestId('combobox-input');
      simulateInputChange(input, searchTerm);
      
      await userEvent.type(input, '{arrowdown}');
      await userEvent.type(input, '{arrowup}');

      // Wait for the input to have focus
      await waitFor(() => {
        expect(document.activeElement).to.eq(input);
      });
    });`

<img width="1077" alt="Screenshot 2024-09-23 at 3 54 26 PM" src="https://github.com/user-attachments/assets/f52469ba-5770-4ab1-801a-77b6a459c0e5">